addons/html_editor/static/src/core/delete_plugin.js

```diff
@@ -600,32 +600,29 @@ export class DeletePlugin extends Plugin {
     // If joined, the range is collapsed to start.
     // Returns the updated range.
     joinFragments(range) {
-        const { joinableLeft, joinableRight } = this.getJoinableFragments(range);
         const { commonAncestorContainer: commonAncestor } = range;
+        const { joinableLeft, joinableRight } = this.getJoinableFragments(range);
+        const [left, right] = [joinableLeft.node, joinableRight.node];
+        const scenario = joinableLeft.type + " + " + joinableRight.type;
 
-        if (!joinableLeft || !joinableRight) {
-            // Nothing to join. Consider it joined already.
-            return this.collapseRange(range);
+        let joined = false;
+        switch (scenario) {
+            case "block + block":
+                joined = this.joinBlocks(left, right, commonAncestor);
+                break;
+            case "block + inline":
+                joined = this.joinInlineIntoBlock(left, right, commonAncestor);
+                break;
+            case "inline + block":
+                joined = this.joinBlockIntoInline(left, right, commonAncestor);
+                break;
+            case "inline + inline":
+            default:
+                // Nothing to join. Consider it joined already.
+                joined = true;
         }
 
-        this.canBeMerged(joinableLeft, joinableRight, commonAncestor);
-        if (
-            joinableLeft.isBlock &&
-            joinableRight.isBlock &&
-            !joinableLeft.isUnmergeable &&
-            !joinableRight.isUnmergeable
-        ) {
-            this.mergeBlocks(joinableLeft.node, joinableRight.node, commonAncestor);
-        } else if (joinableLeft.isBlock && !joinableRight.isBlock && !joinableLeft.isUnmergeable) {
-            this.joinInlineIntoBlock(joinableLeft.node, joinableRight.node);
-        } else if (joinableRight.isBlock && !joinableLeft.isBlock && !joinableRight.isUnmergeable) {
-            this.joinBlockIntoInline(joinableLeft.node, joinableRight.node, commonAncestor);
-        } else if (joinableRight.isUnmergeable || joinableLeft.isUnmergeable) {
-            // Both nodes are inline but at least one of them is unmergeable.
-            // Cannot join. Keep range unchanged.
-            return range;
-        }
-        return this.collapseRange(range);
+        return joined ? this.collapseRange(range) : range;
```

Comment:
I had some failing tests when I applied "collapseRange" on 2 inlines if one of them was "unMergeable", so I kept the range as is in that case without thinking about it too much, so I think these tests will fail too with this code? We'll see :)

Reply:
nevermind seems like it's green :)